### PR TITLE
Add joint feedback checker

### PR DESCRIPTION
**Pull request by efernandez**
This PR provides:
1. Simplify the state initialization using helper static functions
2. Simplify the error calculation using helper static functions
3. Joint Feedback Checker **(this is the main contribution)** for:
- **convergence** of the actual state to the desired one, using a `settling_time` and `error_band` parameters
- **continuity** of the actual state wrt the previous one, using a `max_change` parameter
- **discrepancy** between actual and actual estimated (as velocity estimated from position) state, using an `error_estimate` parameter

@servos @afakihcpr

### CODE REVIEW

**Comment by efernandez** [comment]
All tests added pass.

Note this doesn't change any previous functionality or behavior. That would come in other PRs, but that's not urgent ATM.

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 1]
```diff
@@ -120,6 +120,76 @@ static bool getWheelRadius(
...
+      const std::vector<double>& actual)
+  {
+    const size_t n = err.size();
+
+    ROS_ASSERT(n == desired.size() && n == actual.size());
+
+    for (size_t i = 0; i < n; ++i)
+    {
+      err[i] = desired[i] - actual[i];
+    }
```
Comment (afakihcpr) [inline diff_drive_controller/src/diff_drive_controller.cpp]: You could do instead:
`std::transform (desired.begin(), desired.end(), actual.begin(), err.begin(), std::minus<double>());`

Comment (efernandez) [inline diff_drive_controller/src/diff_drive_controller.cpp]: Done. I've also made it templated.

File: diff_drive_controller/src/joint_feedback_checker.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const double max_change
+  )
+    : settling_time_(settling_time)
+    , error_band_(error_band)
+    , error_estimate_(error_estimate)
+    , max_change_(max_change)
+  {
+  }
+
+  bool JointFeedbackChecker::isSettled(const double dt) const
```
Comment (afakihcpr) [inline diff_drive_controller/src/joint_feedback_checker.cpp]: maybe inline this

**Comment by afakihcpr** [comment]
:+1:

File: diff_drive_controller/src/joint_feedback_checker.cpp (hunk 2 above)
Comment (efernandez) [inline diff_drive_controller/src/joint_feedback_checker.cpp]: Done. I've made the class header only, since most of the functions are really small/trivial.

**Comment by efernandez** [comment]
ping @servos

**Comment by servos** [comment]
LGTM

**Pull request merged**